crucible/src/Lang/Crucible/Backend.hs

```diff
@@ -613,6 +626,59 @@ readPartExpr bak (PE p v) msg = do
   addAssertion bak (LabeledPred p (SimError loc msg))
   return v
 
+
+-- | Run the CHC solver on the current proof obligations, and return the
+-- solution as a substitution from the uninterpreted functions to their
+-- definitions.
+runCHC ::
+  (IsSymBackend sym bak, sym ~ ExprBuilder t st fs, MonadIO m) =>
+  bak ->
+  [SomeSymFn sym] ->
+  m (MapF (SymFnWrapper sym) (SymFnWrapper sym))
+runCHC bak uninterp_inv_fns  = liftIO $ do
+  let sym = backendGetSym bak
+
+  implications <- proofObligationsAsImplications bak
+  clearProofObligations bak
+
+  -- log to stdout
+  let logData = defaultLogData
+        { logCallbackVerbose = \_ -> putStrLn
```

Comment:
Do we really want to always log things to `stdout`? I would expect the user to have to specify this manually if they want a higher-than-normal log setting.

Reply:
copy-pasta